### PR TITLE
Add `pre-commit` for automated formatting and linting.

### PR DESCRIPTION
**Pull request by fabiocaccamo**
Add (and run) [pre-commit](https://pre-commit.com/) with the following hooks:
- `black`
- `isort`
- `flake8`

Partially solves https://github.com/googlefonts/fontra/issues/243

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[tool.black]
+line-length = 89
+include = '\.pyi?$'
+exclude = '''
+/(
```
Comment (justvanrossum) [inline pyproject.toml:5]: Is this really needed? Doesn't it use .gitignore by default?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[tool.black]
+line-length = 89
```
Comment (justvanrossum) [inline pyproject.toml:2]: Why override this? I'm normally happy with black's default of 88.

**Comment by justvanrossum** [comment]
How do I test the behavior of this PR? If I check out the branch, what do I do? Do a test commit?

**Comment by fabiocaccamo** [comment]
Just run `pre-commit run -a` before committing stuff and all checks will run on all (`-a`) files.

File: pyproject.toml (hunk 2 above)
Comment (fabiocaccamo) [reply pyproject.toml:2]: The current value use by the standard library is 79:
`https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html`

I used the same value for both `flake8` and `Black`.

File: pyproject.toml (hunk 1 above)
Comment (fabiocaccamo) [reply pyproject.toml:5]: I'm not sure about how this works when executed by `pre-commit`, but I would say: "explicit is better than implicit"

File: pyproject.toml (hunk 2 above)
Comment (justvanrossum) [reply pyproject.toml:2]: Stdlib is indeed 79, black is 88. We configured prettier to 88, too. So why 89 here?

File: pyproject.toml (hunk 1 above)
Comment (justvanrossum) [reply pyproject.toml:5]: I prefer to not include this cruft if it is taken from .gitignore (which it should). I do like "explicit is better than implicit" but I dislike redundant stuff more.

File: pyproject.toml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [tool.black]
-line-length = 89
+line-length = 88
```
Comment (justvanrossum) [inline pyproject.toml:2]: I would not set it at all, since it's the default.

File: .github/workflows/python-app.yml [hunk 4]
```diff
@@ -56,6 +56,10 @@ jobs:
         pip install -r requirements.txt
         pip install pytest
 
+    - name: Run pre-commit
```
Comment (justvanrossum) [inline .github/workflows/python-app.yml:59]: We may have to rethink this: the action is currently only run manually, not on every push. Also, you're not installing requirements-dev.txt.

Comment (justvanrossum) [reply .github/workflows/python-app.yml:59]: And where I wrote "manually", I meant "manually" and "scheduled once per day". Good enough actually, for Fontra-Pak.

Comment (fabiocaccamo) [reply .github/workflows/python-app.yml:59]: Should we run it also on `push`?

File: pyproject.toml (hunk 3 above)
Comment (justvanrossum) [reply pyproject.toml:2]: > True, but if in a future version of Black the default will change, the CI will not pass anymore, do you want to remove it anyway?

All hell will break loose. I don't think black can afford to change the default.

Comment (fabiocaccamo) [reply pyproject.toml:2]: True!

File: .github/workflows/python-app.yml (hunk 4 above)
Comment (justvanrossum) [reply .github/workflows/python-app.yml:60]: Not this workflow, it's too expensive. Let's leave things for now. Potentially we could have a separate workflow to be run on push that does a light test, but let's not go too deep now.

**Pull request merged**